util/update_logdefs.ts

```diff
@@ -0,0 +1,441 @@
+          // in that case exit the loop & move on
+          if (idLine === 0)
+            idLine = lineNum;
+          else
+            break;
+        } else if (idLine > 0 && line.includes('netRegex: {')) {
+          regexLine = lineNum;
+          break;
+        }
+      }
+
+      if (idLine === 0) {
+        // Might not have been able to find an id: line because the trigger may be
+        // generated dynamically - see aloalo_island.  In that case, just string-search
+        // for the naked id by itself (it has to be there somewhere).
+        lineNum = 0;
+        for (const line of lines) {
+          ++lineNum;
+          if (line.includes(`'${escapedId}'`)) {
+            idLine = lineNum;
+            break;
+          }
+        }
+        if (idLine === 0) {
+          console.error(`ERROR: Could not find trigger '${id}' in ${file}`);
+          continue;
+        } else
+          regexLine = undefined; // don't try to add lines to the excerpt
+      }
+
+      // if we found the id but not the regex, just capture the two lines after the id line
+      regexLine = regexLine === 0 ? idLine + 2 : regexLine;
+
+      if (!this.isLogDefinitionName(type)) {
+        console.error(`ERROR: Missing log def for ${type} in ${file} (line: ${idLine})`);
+        continue;
+      } else if (
+        this.logDefsNoInclude.includes(type) ||
+        this.logDefsNeverInclude.includes(type)
+      )
+        (this.matches[type] ??= []).push({
+          filename: file.replace(`${this.projectRoot}/`, ''),
+          excerptStartLine: idLine,
+          excerptStopLine: regexLine,
+        });
+    }
+  }
+
+  parseTimelineFile(file: string): void {
+    const contents = fs.readFileSync(file).toString();
+    const entries = new TimelineTypeExtractor(contents).entries;
+
+    if (entries === undefined) {
+      console.error(`ERROR: Could not find timeline sync entries in ${file}`);
+      return;
+    }
+
+    for (const [type, lineNums] of Object.entries(entries)) {
+      if (!this.isLogDefinitionName(type))
+        console.error(
+          `ERROR: Missing log def for ${type} in ${file} (line: ${lineNums[0] ?? '?'})`,
+        );
+      else if (
+        this.logDefsNoInclude.includes(type) ||
+        this.logDefsNeverInclude.includes(type)
+      ) {
+        for (const lineNum of lineNums) {
+          (this.matches[type] ??= []).push({
+            filename: file.replace(`${this.projectRoot}/`, ''),
+            excerptStartLine: lineNum,
+          });
+        }
+      }
+    }
+  }
+
+  updateNetLogDefsFile(): void {
+    if (this.logDefsNoInclude.length === 0)
+      return;
+
+    const contents = fs.readFileSync(path.posix.join(this.projectRoot, netLogDefsFile)).toString();
+    const lines = contents.split(/\r*\n/);
+    const fileRegex = {
+      inConst: /^const latestLogDefinitions = {/,
+      inLogDef: /^ {2}(\w+): \{/,
+      outLogDef: /^ {2}\},/,
+      outConst: /^} as const;/,
+    };
```

Comment:
I'm still not really happy with how this function works in general, even after a second review, but I can't really see a better way to handle it.

My instincts are telling me that it should be possible to do something like:
1. Import the current `latestLogDefinitions`
2. Clone the object, stripping out the `const`
3. Update the cloned object to have the new properties
4. `JSON.stringify` it
5. Replace the contents in the file

But I can't really figure out the process in more detail than that right now, so I'm not sure if it would actually work.

Consider this a general comment/thought rather than a request for change.

Reply:
I share the motivation, and I think that conceptually works, but the problem is that it will blow away any inline comments in the const, as they won't be imported.  I couldn't figure out a way around that wouldn't involve regex'ing all comments and then trying to add them back in after `JSON.stringify`, and that would be even more gross...